src/backend/distributed/executor/multi_real_time_executor.c

```diff
@@ -246,6 +264,37 @@ MultiRealTimeExecute(Job *job)
 }
 
 
+static bool
+checkIfSizeLimitIsExceeded(Task *task)
+{
+	struct stat stat;
+
+	StringInfo jobDirectoryName = MasterJobDirectoryName(task->jobId);
+	StringInfo taskFilename = TaskFilename(jobDirectoryName, task->taskId);
+
+	char *filename = taskFilename->data;
+	int fileFlags = (O_RDONLY);
+	int fileMode = (S_IRUSR);
+
+	int32 fileDescriptor = BasicOpenFilePerm(filename, fileFlags, fileMode);
+
+	uint64 MaxIntermediateResultInBytes = MaxIntermediateResult * 1024L;
+
+	int statOK = fstat(fileDescriptor, &stat);
+	close(fileDescriptor);
+	if (statOK == 0 && SetResultLimit && MaxIntermediateResult > -1)
+	{
+		currentIntermediateResult += (uint64) stat.st_size;
+		if (currentIntermediateResult > MaxIntermediateResultInBytes)
```

Comment:
If we take the file size, then we're allowing up to 32GB per subquery, so we could easily fill up a disk. What if we keep a global counter across all tasks?

Reply:
Actually, the currentIntermediateResult is externed and it is reset when the transaction is over. So in the same transaction we continue to increase it? Isn't this the same with a global counter?

Reply:
It seems we're currently increasing `currentIntermediateResult` by the file sizes of all tasks at every iteration of the real-time executor (every `remote_task_check_interval`).

That's fixable, but it would be simpler and much more efficient to just get the number of bytes copied from MultiClientCopyData and add that to the counter at:
https://github.com/citusdata/citus/blob/master/src/backend/distributed/executor/multi_real_time_executor.c#L665